src/Core/Validation/ValidationException.php

```diff
@@ -71,4 +85,21 @@ public function getResult()
     {
         return $this->result;
     }
+
+    /**
+     * Whether to show additional information in the error message depending on the context
+     *
+     * We do not want this to show this in a context where it's easy to know the record and value that
+     * triggered the error e.g. form submission, API calls, etc
+     */
+    private function doShowAdditionalInfo()
+    {
+        if (Director::is_cli()) {
+            return true;
+        }
+        // e.g. dev/build
+        if (is_a(Controller::curr(), DevelopmentAdmin::class)) {
+            return true;
+        }
```

Comment:
Why specifically that controller? I feel like it makes more sense to have a configurable deny list, and say "If the controller is in the deny list, return false" (e.g. `LeftAndMain` and maybe even up to `FormSchemaController` should be in the deny list, but if the current controller is front-end or custom, we should assume the extra context is needed unless developers say otherwise)

Reply:
The point of this PR is to add additional info to the error message _string_, e.g. `fieldName`.  `fieldName` is already part of ValidationResult.  However in a CLI or dev/build context, you simply get the error message outputted to the terminal/screen - you don't see the rest of the metadata in the ValidationResult object.

We don't want the fieldname being suffixed to the error message string in a UX context because then you'd end up with a validation message like "Invalid email address - fieldname: Email" showing up underneath the Email field which already has a label "Email"

Reply:
Looking at [LeftAndMain::handleRequest()](https://github.com/silverstripe/silverstripe-admin/blob/2/code/LeftAndMain.php#L856) - it's using $e->getMessage(), which a native PHP method for Exception.   In ValidationException, whose parent class is Exception, we're calling parent::__construct($exceptionMessage, $code);

LeftAndMain::handleRequest is obviously a pretty central place, so it seems pretty likely this additional info will popup on the frontend where we don't intend it to.

I've upgraded the the allow list to make it configurable

Reply:
I don't know of any scenarios where that actually gets hit - for form submission requests the exception gets caught inside [`FormRequestHandler::httpSubmission()`](https://github.com/silverstripe/silverstripe-framework/blob/873d552084a2b784b285dd01e914b6c33db98a94/src/Forms/FormRequestHandler.php#L247-L252) where the messages are taken from the `ValidationResult` - _not_ the exception message.

The only time the `catch` in `LeftAndMain::handleRequest()` comes into play is I guess if you're doing something AJAX-y which doesn't have anything to do with a form, in which case that extra context is probably going to be useful and should be included in the displayed message.

Reply:
Something AJAX-y could be literally anything, from submitting an entire form, to a submitting a single field, or something no even related to submitting data.  We cannot simply assume that something that's a debug message aimed at developers will look correct in any context. It could result in a red toast notification aimed at content editors, or something different. We just don't know, so this should be left as is.

Reply:
Have discussed offline and backlogged a new card https://github.com/silverstripe/silverstripe-framework/issues/11599